applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -125,6 +132,32 @@ static void le_audio_rx_data_handler(uint8_t const *const p_data, size_t data_si
 	/* Since the audio datapath thread is preemptive, no actions on the
 	 * FIFO can happen whilst in this handler.
 	 */
+	bool data_size_mismatch = false;
+	static struct rx_stats rx_stats[AUDIO_CH_NUM];
+
+	rx_stats[channel_index].recv_cnt++;
+	if (data_size != desired_data_size && bad_frame != true) {
+		data_size_mismatch = true;
+		rx_stats[channel_index].data_size_mismatch_cnt++;
+	}
+
+	if (bad_frame) {
+		rx_stats[channel_index].bad_frame_cnt++;
+	}
+
+	for (int i = 0; i < AUDIO_CH_NUM; i++) {
+		if ((rx_stats[i].recv_cnt % 100) == 0 && rx_stats[i].recv_cnt) {
+			LOG_DBG("ISO RX SDUs: Ch: %d Total: %d Bad: %d Size mismatch %d", i,
+				rx_stats[channel_index].recv_cnt,
```

Comment:
```suggestion
				rx_stats[i].recv_cnt,
```

Same for the two below

Reply:
The for loop is now removed, so this is not relevant any more. That is intended, right?

Reply:
Yes, it was printing incorrectly with the previous approach.